Public/Src/Pips/Dll/Operations/Process.cs

```diff
@@ -596,6 +600,16 @@ public sealed partial class Process : Pip
         /// </summary>
         public bool AllowPreserveOutputs => (ProcessOptions & Options.AllowPreserveOutputs) != 0;
 
+        /// <summary>
+        /// File/directory output paths that are preserved if <see cref="AllowPreserveOutputs"/> is enabled. 
+        /// </summary>
+        /// <remarks>
+        /// If the list is empty, all file and directory outputs are preserved. If the list is not empty,
+        /// only given paths are preserved and the rest is deleted.
+        /// </remarks>
+        [SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
+        public ReadOnlyArray<AbsolutePath> PreserveOutputWhitelist { get; }

```

Comment:
>public ReadOnlyArray<AbsolutePath> PreserveOutputWhitelist { get; } [](start = 8, length = 67)

Should this affect fingerprint? 

Do we have unit test where we remove or add to the whitelist?

Reply:
Because “preserveOutputs” option does not affect the fingerprint, i thought that it makes sense that the whitelist also has no effect. 

I will add a test to modify the whitelist and expect a cache hit or miss depending on the availability of the output.

Reply:
Because “preserveOutputs” option does not affect the fingerprint, i thought that it makes sense that the whitelist also has no effect. 

I will add a test to modify the whitelist and expect a cache hit or miss depending on the availability of the output.